doc/manual/src/contributing/hacking.md

```diff
@@ -71,21 +98,44 @@ To install it in `$(pwd)/outputs` and test it:
 [nix-shell]$ make install
 [nix-shell]$ make installcheck -j $NIX_BUILD_CORES
 [nix-shell]$ ./outputs/out/bin/nix --version
-nix (Nix) 3.0
+nix (Nix) 2.12
 ```
 
-If you have a flakes-enabled Nix you can replace:
+To build Nix for the current operating system and CPU architecture use
 
 ```console
-$ nix-shell
+$ nix-build
 ```
 
-by:
+This will build `packages.x86_64-linux.default` attribute defined in the [flake.nix](./flake.nix)
+file.
+
+To build for other platforms, use one of the following platform identifiers:
+
+- `aarch64-linux`
+- `i686-linux`
+- `x86_64-darwin`
+- `x86_64-linux`
+
+Example:
 
 ```console
-$ nix develop
+$ nix-build -A packages.x86_64-linux.default
 ```
 
+## Developing
+
+`clangd` and `bear` are installed by default in the `devShell`. To use LSP with your editor, after running the above `configure` steps,
+run
+
+```console
+make clean && bear -- make -j$NIX_BUILD_CORES install
+```
+
+to generate the neccesary `compile_commands.json` file.
```

Comment:
```suggestion
The `clangd` LSP server is installed by default on the `clang`-based `devShell`s (see above for using them instead of the default one).

To use the LSP with your editor, you first need to [set it up](https://clangd.llvm.org/installation#project-setup) file by running:

```console
make clean && bear -- make -j$NIX_BUILD_CORES install
```

(With an extra “```” at the end, but these don't play well with github's suggestions